python/openchange/provision.py

```diff
@@ -920,6 +1096,50 @@ def openchangedb_deprovision(names, lp, mapistore=None):
     openchangedb.remove()
 
 
+# migration methods for openchange directory
+def directory_migrate(lp, uri=None, version=None, extra=None):
+    provisioned = False
+    try:
+        check_not_provisioned(extra['names'], lp, extra['creds'])
+    except:
+        provisioned = True
+    if not provisioned:
+        raise Exception("OpenChange is not provisioned, provision it instead of upgrading it")
+
```

Comment:
I agree with you but I explained here in order to have a global picture. I think the scope is correct but we have to know we do not have symmetric behaviour in Database and Directory. Thus wrong assumptions may appear in the future.